### ISSUE TITLE
Add a Makefile

### ISSUE THREAD

**Issue by jt-nti** [hyperledger/fabric-contract-api-go#58]
It's not immediately obvious (to me!) how to build and test this repo so a Makefile might be useful

**Comment by shashank-priyadarshi** [comment]
Hi @jt-nti , can I take this up? Could you please also tell me what is this Makefile supposed to do!

**Comment by jt-nti** [comment]
Hi @shashank-priyadarshi, I would check with the current maintainers @mbwhite and @bestbeforetoday before starting any work but the idea was that it might be helpful to be able to build the project using make, for example the [fabric-gateway](https://github.com/hyperledger/fabric-gateway) project has a [Makefile](https://github.com/hyperledger/fabric-gateway/blob/main/Makefile).

**Comment by bestbeforetoday** [comment]
I think the Makefile should define the set of tasks (targets) that you might reasonably want to run when doing development on this project. A good guide for candidate tasks are the things that are run in the automated GitHub Actions [workflows](https://github.com/hyperledger/fabric-contract-api-go/tree/main/.github/workflows). Once Makefile targets for tasks are implemented, the GitHub Actions workflows can be modified to run the Makefile targets. Developers working on the code can see (in the Makefile) the set of useful tasks, and use Make to run them locally in exactly the same way that they will be run in the automated build.

The [fabric-gateway Makefile](https://github.com/hyperledger/fabric-gateway/blob/main/Makefile) is much more complex that what it likely needed for this repository, since it deals with multiple language implementations. It is probably a good model to follow though, with targets like `unit-test`, `scenario-test`, `lint`, `scan` and so on.

Optionally, there might be an opportunity to improve on to way some tasks are implemented too. Perhaps golangci-lint would be a good alternative to a script explicitly running **gofmt**, **goimports** and **go vet**, which relies on tools being separately installed. This could certainly be done as a separate pull request though, once a Makefile is in place.

If you are still interested in adding a Makefile, I can assign this issue to you.

**Comment by shashank-priyadarshi** [comment]
Sure @bestbeforetoday I can work on this!

**Comment by shashank-priyadarshi** [comment]
Hi @bestbeforetoday , what are the tasks that this Makefile should essentially accomplish.
With reference to the [fabric-gateway Makefile](https://github.com/hyperledger/fabric-gateway/blob/main/Makefile) and @jt-nti 's comment at the beginning of the thread, I can think of the following:

- basic-checks - performs basic checks like license, spelling, trailing spaces and linter
- checks - runs all non-integration tests/checks
- desk-check - runs linters and verify to test changed packages
- docs - builds the documentation in html format
- gotools - installs go tools like golint
- integration-test-prereqs - setup prerequisites for integration tests
- integration-test - runs the integration tests
- license - checks go source files for Apache license header
- linter - runs all code checks
- verify - runs unit tests for only the changed package tree

Please let me know if this could be a starting point, and what other things should I include?

**Comment by bestbeforetoday** [comment]
A Makefile was added as part of PR #134.
